front/app/modules/commercial/insights/admin/containers/Insights/Edit/Categories/index.tsx

```diff
@@ -358,14 +427,25 @@ const Categories = ({
               onClick={selectCategory(category.id)}
             >
               <div>{category.attributes.name}</div>
-              <div data-testid="insightsCategoryCount">
+              <div
+                className="buttonCountText"
+                data-testid="insightsCategoryCount"
+              >
                 {category.attributes.inputs_count}
               </div>
-            </CategoryButton>
+              <div
+                className="buttonIcon"
+                role="button"
```

Comment:
Why does the icon need role button inside a button? A div with role button (or role button in general) is not the best for a11y.

Reply:
Yes, I had this exact argument with Antoine when he suggested this change and I also dislike it for accessibility reasons. 

Here, we have a delete icon that only appears on hover (also bad for a11y!) and the idea is that users expect to have the action there. Since there is an accessible way to do the same action (via a menu elsewhere in the screen), it's not so bad... It's similar to how gmail handles the same use case with the icons that appear on hover.